File: #1-25 (C, Java, Python)/Java/Euler2.java
//Euler2 ~ Even Fibonacci numbers
import java.util.concurrent.TimeUnit;

public class Euler2
{
    public static void main(String[] args) throws InterruptedException
	{
		long startTime = System.nanoTime();
		
		int sum = 2;
		int f1 = 1;
		int f2 = 2;
		int f3 = 0;
		
		do
		{
			f3 = f1 + f2;
			f1 = f2;
			f2 = f3;
			
			if((f3%2)==0)
			{
				sum = sum + f3;
			}
		} while(f3<4000000);
		
		long endTime = System.nanoTime();
		long timeElapsed = endTime - startTime;
		System.out.println("Solution: " + sum + "\n");
		System.out.println("Program completed in " + timeElapsed/1000000000 + " seconds.");
    }
}

File: #1-25 (C, Java, Python)/Python/Euler2.py
#Euler2 ~ Even Fibonacci numbers
import timeit
start = timeit.default_timer()

s  = 2
f1 = 1
f2 = 2
f3 = 0

while f3 < 4000000:
    f3 = f1 + f2
    f1 = f2
    f2 = f3
    if((f3%2) == 0):
        s = s + f3

stop = timeit.default_timer()
print("Solution: ",s,"\n")
print("Program completed in","{0:.3f}".format(stop - start), "seconds.")